File: homework01-go/vigenere/vigenere.go
package vigenere

func EncryptVigenere(plaintext string, keyword string) string {
	var ciphertext string

	// PUT YOUR CODE HERE
    ciphertext=str()
    tro=0
    for i in range(0,len(plaintext)):
        if plaintext[i] >= 'A':
            if plaintext[i]<='Z':
                t=ord(plaintext[i])+ord(keyword[tro])-ord('A')
                if t > ord('Z'):
                    t-=26
                ciphertext+=chr(t)
        if plaintext[i] >= 'a':
            if plaintext[i]<='z':
                t=ord(plaintext[i])+ord(keyword[tro])-ord('a')
                if t > ord('z'):
                    t-=26
                ciphertext+=chr(t)
        tro+=1
        if(tro== len(keyword)):
            tro=0
	return ciphertext
}

func DecryptVigenere(ciphertext string, keyword string) string {
	var plaintext string

	// PUT YOUR CODE HERE
    plaintext=str()
    tro=0
    for i in range(0,len(ciphertext)):
        if ciphertext[i] >= 'A':
            if ciphertext[i]<='Z':
                t=ord(ciphertext[i])-ord(keyword[tro])+ord('A')
                if t < ord('A'):
                    t+=26
                plaintext+=chr(t)
        if ciphertext[i] >= 'a':
            if ciphertext[i]<='z':
                t=ord(sciphertext[i])-ord(keyword[tro])+ord('a')
                if t < ord('a'):
                    t+=26
                plaintext+=chr(t)
        tro+=1
        if(tro== len(keyword)):
            tro=0
	return plaintext
}

File: homework01/vigenere.py
a = []
b = []
for m in range(65, 91):
    a.append(chr(m))
for n in range(97, 123):
    b.append(chr(n))


def encrypt_vigenere(plaintext: str, keyword: str) -> str:
    d = len(plaintext) // len(keyword)
    keyword = list(keyword)
    keyword1 = list(range(len(keyword)))
    for i in range(len(keyword)):
        for j in range(len(a)):
            if keyword[i] == a[j]:
                keyword1[i] = j
            if keyword[i] == b[j]:
                keyword1[i] = j
    tran = keyword1 * (d + 1)
    ciphertext = list(plaintext)
    for i in range(len(plaintext)):
        for j in range(len(a)):
            if plaintext[i] == a[j]:
                if j + tran[i] < 26:
                    ciphertext[i] = a[j + tran[i]]
                else:
                    ciphertext[i] = a[j + tran[i] - 26]
            if plaintext[i] == b[j]:
                if j + tran[i] < 26:
                    ciphertext[i] = b[j + tran[i]]
                else:
                    ciphertext[i] = b[j + tran[i] - 26]
    ciphertext = ''.join(ciphertext)
    return ciphertext


def decrypt_vigenere(ciphertext: str, keyword: str) -> str:
    d = len(ciphertext) // len(keyword)
    keyword = list(keyword)
    keyword1 = list(range(len(keyword)))
    for i in range(len(keyword)):
        for j in range(len(a)):
            if keyword[i] == a[j]:
                keyword1[i] = j
            if keyword[i] == b[j]:
                keyword1[i] = j
    tran = keyword1 * (d + 1)
    plaintext = list(ciphertext)
    for i in range(len(ciphertext)):
        for j in range(len(a)):
            if ciphertext[i] == a[j]:
                if j < tran[i]:
                    plaintext[i] = a[j + 26 - tran[i]]
                else:
                    plaintext[i] = a[j - tran[i]]
            if ciphertext[i] == b[j]:
                if j < tran[i]:
                    plaintext[i] = b[j + 26 - tran[i]]
                else:
                    plaintext[i] = b[j - tran[i]]
    plaintext = ''.join(plaintext)
    return plaintext
